proto/message_contents/public_key.proto

```diff
@@ -9,6 +9,53 @@ import "message_contents/signature.proto";
 option go_package = "github.com/xmtp/proto/v3/go/message_contents";
 option java_package = "org.xmtp.proto.message.contents";
 
+// An unsigned public key used by libxmtp
+message VoodooUnsignedPublicKey {
+    uint64 created_ns = 1;
+
+    oneof union {
+        VodozemacCurve25519 curve25519 = 2;
+    }
+
+    // A Vodozemac curve25519 key serialized via serde
+    message VodozemacCurve25519 {
+        bytes bytes = 1;
+    }
+}
+
+// A key linked to an XMTP account (e.g. signed by a wallet)
+// The purpose of the key is encoded in the signature
+message VoodooAccountLinkedKey {
+    VoodooUnsignedPublicKey key = 1;
+    // TODO AccountLinkedSignature signature = 2;
+}
+
+// A key linked to a device (e.g. signed by a device identity key)
+// The purpose of the key is encoded in the signature
+message VoodooDeviceLinkedKey {
+    VoodooUnsignedPublicKey key = 1;
+    // TODO DeviceLinkedSignature signature = 2;
+}
+
+// A bundle of one time keys uploaded by a client
+message VoodooOneTimeKeyTopupBundle {
+    VoodooAccountLinkedKey identity_key = 1;
+    repeated VoodooDeviceLinkedKey one_time_keys = 2;
+}
+
+// A fallback key uploaded by a client
+message VoodooFallbackKeyRotation {
```

Comment:
I would love more verbose comments! For example, explaining way it's called a Rotation instead of just `VoodooFallbackKey`